Tools/LyTestTools/ly_test_tools/o3de/editor_test.py

```diff
@@ -200,667 +144,69 @@ class EditorBatchedTest(EditorSharedTest):
-        test owns it.
-        :workspace: The workspace object in case an AssetProcessor object needs to be created
-        :editor_test_data: The test data from calling editor_test_data()
-        :return: None
-        """
-        try:
-            # Start-up an asset processor if we are not already managing one
-            if editor_test_data.asset_processor is None:
-                if not process_utils.process_exists("AssetProcessor", ignore_extensions=True):
-                    editor_utils.kill_all_ly_processes(include_asset_processor=True)
-                    editor_test_data.asset_processor = AssetProcessor(workspace)
-                    editor_test_data.asset_processor.start()
-                else:  # If another AP process already exists, do not kill it as we do not manage it
-                    editor_utils.kill_all_ly_processes(include_asset_processor=False)
-            else:  # Make sure existing asset processor wasn't closed by accident
-                editor_test_data.asset_processor.start()
-        except Exception as ex:
-            editor_test_data.asset_processor = None
-            raise ex
-
-    def _setup_editor_test(self, editor: ly_test_tools.launchers.platforms.base.Launcher,
-                           workspace: ly_test_tools._internal.managers.workspace.AbstractWorkspaceManager,
-                           editor_test_data: TestData) -> None:
-        """
-        Sets up an editor test by preparing the Asset Processor, killing all other O3DE processes, and configuring
-        :editor: The launcher Editor object
-        :workspace: The test Workspace object
-        :editor_test_data: The TestData from calling editor_test_data()
-        :return: None
-        """
-        self._prepare_asset_processor(workspace, editor_test_data)
-        editor_utils.kill_all_ly_processes(include_asset_processor=False)
-        editor.configure_settings()
-
-    @staticmethod
-    def _get_results_using_output(test_spec_list: list[EditorTest], output: str, editor_log_content: str) -> dict[str, Result.ResultType]:
-        """
-        Utility function for parsing the output information from the editor. It deserializes the JSON content printed in
-        the output for every test and returns that information.
-        :test_spec_list: The list of EditorTests
-        :output: The Editor from Editor.get_output()
-        :editor_log_content: The contents of the editor log as a string
-        :return: A dict of the tests and their respective Result objects
+    # Maximum time (seconds) for waiting for a crash file to finish being dumped to disk
+    _TIMEOUT_CRASH_LOG = 20
+    # Return code for test failure
+    _TEST_FAIL_RETCODE = 0xF
+    # Test class to use for single test collection
+    single_test_class = EditorSingleTest
+    # Test class to use for shared test collection
+    shared_test_class = EditorSharedTest
+
+    @pytest.mark.parametrize("crash_log_watchdog", [("raise_on_crash", False)])
```

Comment:
From this point onward, there appears to be a decent amount of code related to marking and executing tests that should be possible to hoist into the parent class. Was there a reason not to?

This currently means a lot of code gets duplicated in both EditorTest and MaterialEditorTest. The only thing I expected to remain in EditorTest was data, and perhaps a couple functions for setup/cleanup/reporting hooks.

Reply:
I believe it had to do with the collection order with pytest. Depending on where certain decorators are placed it can make the collection order collect in the order not intended since we're hijacking the built-in decorator based pytest collection using `AbstractTestClass(pytest.Class)` + `pytest_pycollect_makeitem()`

I can re-explore changing it again and report the specific error encountered here should I observe it again.

Reply:
Upon researching this further I feel it is a re-factor task ask to tidy up / clean up the existing execution code. I don't want to do any in-depth re-factoring work in this PR but simply want to extend the batched and parallel test tools to support MaterialEditor in addition to Editor. We can do a follow-up task to re-factor things to be more flat and in-line between Editor and MaterialEditor so that `editor_test.py` and `material_editor_test.py` hold less execution code but for the time being I think we shouldn't block this support for such a reason.

Reply:
I would prefer not to duplicate new instances of preexisting problems, but existing problems can be left for cleanup in other tasks (work on which this PR is currently blocking)

Reply:
I will try to merge them in the `multi_test_framework.py` module. New internal tickets are being cut to track this effort.

Reply:
after more research this may be something we're going to have to live with to be able to ship the code until we can change it at a later date

for instance, I have tried sharing attributes between test suite class objects and using `__init__` to construct those attributes on a given test suite class object but this is the kind of error you get when taking that approach (and hence why we don't use `__init__` in the existing code and instead use flat attributes attached to the test suite class objects:
```
AutomatedTesting\Gem\PythonTests\Atom\TestSuite_Main_GPU.py:24
  C:\git\o3de\AutomatedTesting\Gem\PythonTests\Atom\TestSuite_Main_GPU.py:24: PytestCollectionWarning: cannot collect test class 'TestAutomation' because it has a __init__ constructor (from: AutomatedTesting/Gem/PythonTests/Atom/TestSuite_Main_GPU.py)
    class TestAutomation(EditorTestSuite):

-- Docs: https://docs.pytest.org/en/latest/warnings.html
```

Reply:
i will stick with flat attributes and try to convert the code that way

Reply:
this work is now done in the latest update